src/dtclient/kubernetes_settings_test.go

```diff
@@ -0,0 +1,389 @@
+		dtc, err := NewClient(dynatraceServer.URL, apiToken, paasToken, skipCert)
+		require.NoError(t, err)
+		require.NotNil(t, dtc)
+
+		// act
+		actual, err := dtc.(*dynatraceClient).CreateKubernetesSetting(testName, testUID, testScope)
+
+		// assert
+		assert.Error(t, err)
+		assert.Len(t, actual, 0)
+	})
+}
+
+func handleEntitiesRequest(request *http.Request, writer http.ResponseWriter, entities []MonitoredEntity) {
+	if request.Method == http.MethodGet {
+		if !strings.Contains(request.Form.Get("entitySelector"), "type(KUBERNETES_CLUSTER)") {
+			writer.WriteHeader(http.StatusBadRequest)
+			return
+		}
+
+		meResponse := monitoredEntitiesResponse{
+			TotalCount: len(entities),
+			PageSize:   500,
+			Entities:   entities,
+		}
+
+		entitiesResponse, err := json.Marshal(meResponse)
+
+		if err != nil {
+			return
+		}
+
+		writer.WriteHeader(http.StatusOK)
+		writer.Write(entitiesResponse)
+	} else {
+		writeError(writer, http.StatusMethodNotAllowed)
+	}
+}
+
+func handleSettingsRequest(request *http.Request, writer http.ResponseWriter, totalCount int, objectId string) {
+	if request.Method == http.MethodGet {
+		if request.Form.Get("schemaIds") != "builtin:cloud.kubernetes" || request.Form.Get("scopes") == "" {
+			writer.WriteHeader(http.StatusBadRequest)
+			return
+		}
+
+		settingsGetResponse, err := json.Marshal(GetSettingsResponse{TotalCount: totalCount})
+
+		if err != nil {
+			return
+		}
+
+		writer.WriteHeader(http.StatusOK)
+		writer.Write(settingsGetResponse)
+	} else if request.Method == http.MethodPost {
+		if request.Body == nil {
+			writer.WriteHeader(http.StatusBadRequest)
+			return
+		}
+
+		body, err := ioutil.ReadAll(request.Body)
+
+		if err != nil {
+			return
+		}
+
+		var parsedBody []postKubernetesSettingsBody
+		err = json.Unmarshal(body, &parsedBody)
+
+		if err != nil {
+			return
+		}
+
+		var settingsPostResponse []postSettingsResponse
+		settingsPostResponse = append(settingsPostResponse, postSettingsResponse{
+			ObjectId: objectId,
+		})
+
+		settingsPostResponseBytes, err := json.Marshal(settingsPostResponse)
+
+		if err != nil {
+			return
+		}
+
+		writer.WriteHeader(http.StatusOK)
+		writer.Write(settingsPostResponseBytes)
+	} else {
+		writeError(writer, http.StatusMethodNotAllowed)
+	}
+
+}
+
+func dynatraceServerEntitiesHandler(entities []MonitoredEntity, isError bool) http.HandlerFunc {
+	return func(w http.ResponseWriter, r *http.Request) {
+		if isError {
+			writeError(w, http.StatusBadRequest)
+			return
+		}
+
+		w.Header().Set("Content-Type", "application/json")
+
+		if r.FormValue("Api-Token") == "" && r.Header.Get("Authorization") == "" {
+			writeError(w, http.StatusUnauthorized)
+		} else {
+			switch r.URL.Path {
+			case "/v2/entities":
+				handleEntitiesRequest(r, w, entities)
+			default:
+				writeError(w, http.StatusBadRequest)
+			}
+		}
+	}
+}
+
+func dynatraceServerSettingsHandler(totalCount int, objectId string, isError bool) http.HandlerFunc {
```

Comment:
same as before
`mockDynatraceServerSettingsHandler ` or something similar

Reply:
I might have been too quick now with the changes :smile: 
Trying to be correct here - is it a `mock` or a `fake`?

Reply:
you are mocking the dynatrace server or you are faking the dynatrace server ?
dunno which is the technically correct term 😅  (both sound fine to me)
I would go with mock, but not because of any objective reason 😄  (I just like it better)

Reply:
Then let's keep `mock` :slightly_smiling_face: